CenterForOpenScience/osf.io
# [IN-370] Alter AbstractProvider Uniqueness

## Purpose
Allow multiple providers (e.g. of the same organization) across different `type`s to share the same `_id`

## Changes
* Remove `_id` uniqueness constraint
* Add `unique_together` for `_id` and `type`

## QA Notes
Ensure with the admin app that a `PreprintProvider` and a `CollectionProvider` can have the same `_id` set, and that two `*Provider`s of the same type cannot.

## Side Effects
No direct side effects expected. Issues may arise if the mixin is used in other classes and the `unique_together` constraint is not added, or if it's mixed in with `GuidMixin`.

## Ticket
[[IN-370]](https://openscience.atlassian.net/browse/IN-370)
